dev/release-build.sh

```diff
@@ -0,0 +1,268 @@
+
+BASE_DIR=$(pwd)
+
+MVN="mvn"
+PUBLISH_PROFILES="-Pdistribution"
+
+if [ -z "$RELEASE_RC" ]; then
+  RELEASE_RC="rc1"
+fi
+
+if [ -z "$RELEASE_TAG" ]; then
+  RELEASE_TAG="$RELEASE_VERSION-$RELEASE_RC"
+fi
+
+DEST_DIR_NAME="bahir-$DEVELOPMENT_VERSION"
+
+RELEASE_STAGING_LOCATION="https://dist.apache.org/repos/dist/dev/bahir/"
+
+
+echo "  "
+echo "-------------------------------------------------------------"
+echo "------- Release preparation with the following parameters ---"
+echo "-------------------------------------------------------------"
+echo "Goal                ==> $GOAL"
+echo "Git reference       ==> $GIT_REF"
+echo "release version     ==> $RELEASE_VERSION"
+echo "development version ==> $DEVELOPMENT_VERSION"
+echo "rc                  ==> $RELEASE_RC"
+echo "tag                 ==> $RELEASE_TAG"
+echo "  "
+echo "Deploying to :"
+echo $RELEASE_STAGING_LOCATION
+echo "  "
+
+function checkout_code {
+    # Checkout code
+    rm -rf target
+    mkdir target
+    cd target
+    rm -rf bahir
+    git clone https://git-wip-us.apache.org/repos/asf/bahir.git
+    cd bahir
+    git checkout $GIT_REF
+    git_hash=`git rev-parse --short HEAD`
+    echo "Checked out Bahir git hash $git_hash"
+
+    git clean -d -f -x
+    #rm .gitignore
+    #rm -rf .git
+
+    cd $BASE_DIR #return to base dir
+}
+
+if [[ "$RELEASE_PREPARE" == "true" ]]; then
+    echo "Preparing release $RELEASE_VERSION"
+    # Checkout code
+    checkout_code
+    cd target/bahir
+
+    # Build and prepare the release
+    $MVN $PUBLISH_PROFILES release:clean release:prepare -Dgpg.passphrase="$GPG_PASSPHRASE" -DskipTests -DreleaseVersion="$RELEASE_VERSION" -DdevelopmentVersion="$DEVELOPMENT_VERSION" -Dtag="$RELEASE_TAG"
+
+    cd .. #exit bahir
+
+    svn co $RELEASE_STAGING_LOCATION svn-bahir
+    mkdir -p svn-bahir/$RELEASE_VERSION-$RELEASE_RC
+    cp bahir/distribution/target/*.tar.gz svn-bahir/$RELEASE_VERSION-$RELEASE_RC/
+    cp bahir/distribution/target/*.zip    svn-bahir/$RELEASE_VERSION-$RELEASE_RC/
+
+    cd svn-bahir/$RELEASE_VERSION-$RELEASE_RC/
+    for i in *.zip *.gz; do gpg --output $i.asc --detach-sig --armor $i; done
+    for i in *.zip *.gz; do openssl md5 -hex $i | sed 's/MD5(\([^)]*\))= \([0-9a-f]*\)/\2 *\1/' > $i.md5; done
+
+    cd .. #exit $RELEASE_VERSION-$RELEASE_RC/
+
+    svn add $RELEASE_VERSION-$RELEASE_RC/
+    svn ci -m"Apache Bahir $RELEASE_VERSION-$RELEASE_RC"
+
+    cd $BASE_DIR #exit target
+
+    exit 0
+fi
+
+
+if [[ "$RELEASE_PUBLISH" == "true" ]]; then
+    echo "Preparing release $RELEASE_VERSION"
+    # Checkout code
+    checkout_code
+    cd target/bahir
+
+    $MVN $PUBLISH_PROFILES clean install -DskiptTests
+
+    #Deploy default scala 2.11
+    mvn -DaltDeploymentRepository=apache.releases.https::default::https://repository.apache.org/service/local/staging/deploy/maven2 verify gpg:sign install:install deploy:deploy -DskiptTests -Dgpg.passphrase=$GPG_PASSPHRASE $PUBLISH_PROFILES
+
+    #Deploy scala 2.10
```

Comment:
IIRC the maven scala plugin shares a common target directory for all scala versions, so to avoid any discrepancies in classfiles and potenttial "ClassNotFound" errors, I would suggest cleaning here.